src/elements/select/select.ts

```diff
@@ -794,6 +794,11 @@ class SbbSelectElement extends SbbUpdateSchedulerMixin(
   }
 
   private _setNextActiveOption(event: KeyboardEvent, index?: number): void {
+    // Prevent keyboard navigation if all options are disabled
+    if (this._filteredOptions.length === 0) {
```

Comment:
I propose to cache filteredoptions in this method as it is called several times.